### ISSUE TITLE
Copy in tmux for Linux

### ISSUE THREAD

**Issue by ajr-dev** [nicknisi/dotfiles#164]
Since Tmux 3 the syntax for the if else is simpler. You could add copy support for linux.

```
if-shell "uname | grep -q Darwin" {
    bind-key -T copy-mode-vi y send -X copy-pipe-and-cancel 'tmux save-buffer - | reattach-to-user-namespace pbcopy'
    bind C-c run 'tmux save-buffer - | reattach-to-user-namespace pbcopy'
    bind C-v run 'tmux set-buffer $(reattach-to-user-namespace pbpaste); tmux paste-buffer'
} {
    bind-key -T copy-mode-vi v send-keys -X begin-selection
    bind-key -T copy-mode-vi y send -X copy-pipe-and-cancel 'xclip -se c -i'
    unbind -T copy-mode-vi Enter
    bind-key -T copy-mode-vi Enter send -X copy-pipe-and-cancel 'xclip -se c -i'
    bind-key -T copy-mode-vi MouseDragEnd1Pane send-keys -X copy-pipe-and-cancel 'xclip -se c -i'
}
```
Copied from [here](https://www.rockyourcode.com/copy-and-paste-in-tmux/):

    Now you can enter copy mode normally with CTRL+B and [.
    Navigate the copy mode with vi-like-key bindings (u for up, d for down, etc.)
    Hit v to start copying.
    Press y or Enter to copy the text into the tmux buffer. This automatically cancels copy mode.
    Paste into the buffer with <prefix>+P (make sure that it’s uppercase P).

    Or alternatively, use the mouse to copy text after you’ve entered copy mode.

    The above commands use xclip, a Linux command line tool for X11.

xclip doesn't come preinstalled, so it should be added to the brewfile just for linux.
This is how I have my `tmux.conf`, to make it backwards compatible until tmux 2.5.
```
# set vi mode for copy mode
setw -g mode-keys vi
# more settings to make copy-mode more vim-like
unbind [
bind Escape copy-mode
unbind p
bind p paste-buffer
bind -Tcopy-mode-vi v send -X begin-selection
bind-key -T copy-mode-vi C-v send -X rectangle-toggle \; send -X begin-selection

if-shell "uname | grep -q Darwin" " \
    bind-key -T copy-mode-vi y send -X copy-pipe-and-cancel 'tmux save-buffer - | reattach-to-user-namespace pbcopy'; \
    bind C-c run 'tmux save-buffer - | reattach-to-user-namespace pbcopy'; \
    bind C-v run 'tmux set-buffer $(reattach-to-user-namespace pbpaste); tmux paste-buffer'" " \
    bind-key -T copy-mode-vi y send -X copy-pipe-and-cancel 'xclip -selection clipboard -in'"
```

**Comment by nicknisi** [comment]
Yeah, I like this. This also made me realize that reattach-to-user-namespace is missing from the Brewfile and needs to be added as well. Are you interested in sending a PR?

**Comment by ajr-dev** [comment]
I've added the latest changes to my dotfiles and now I can't do anything. I can't even commit.

Missing plugin mafredri/zsh-async
  Cloning from git@github.com:mafredri/zsh-async...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
  [ERROR] git clone errored: 0. aborting...
/home/user/.dotfiles/zsh/prompt.zsh:145: command not found: async_init
/home/user/.dotfiles/zsh/prompt.zsh:146: command not found: async_start_worker
/home/user/.dotfiles/zsh/prompt.zsh:147: command not found: async_register_callback

~ 
precmd:2: command not found: async_job
❯ cd ~/.dotfiles       

~/.dotfiles 
precmd:2: command not found: async_job
❯ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   Brewfile
	modified:   tmux/tmux.conf.symlink

~/.dotfiles 
precmd:2: command not found: async_job
❯ git commit -m "feat(tmux): add copy support on linux"
.git/hooks/pre-commit: 9: shopt: not found

Executing /home/user/.dotfiles/git/hooks/*.pre-commit
.git/hooks/pre-commit: 13: /home/user/.dotfiles/git/hooks/*.pre-commit: not found
.git/hooks/pre-commit: 17: [[: not found

~/.dotfiles 
precmd:2: command not found: async_job
❯ git status                                           
On branch master
Your branch is up to date with 'origin/master'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   Brewfile
	modified:   tmux/tmux.conf.symlink

**Comment by nicknisi** [comment]
zsh_async is installed with the `zfetch` script. It's used to make the git prompt asynchronous and replaces my attempt at doing that myself. It's failing because you must not have a private/public key set up with GitHub. `zfetch` was changed to use SSH in 72d3ad8f.

As for the pre-commit hook, I don't actually have any set up in this repo. But it looks like the hook is not actually running in bash, since `shoot` doesn't seem to exist. The shebang in that file is `#!/usr/bin/env bash`.  Does `usr/bin/env` not exist on your machine?

**Comment by ajr-dev** [comment]
That directory doesn't exist on linux.

**Comment by nicknisi** [comment]
Hmm, what flavor of linux are you running?

<img width="1121" alt="image" src="https://user-images.githubusercontent.com/293805/101246352-5f9ea380-36d8-11eb-897b-de167fe182df.png">

**Comment by ajr-dev** [comment]
That's weird, I'm using Linux Mint, it's based on Ubuntu.

**Comment by nicknisi** [comment]
Fixed in dc09b2f. Thanks!
